File: backend-nestjs/src/modules/department-management/application/mappers/department.mapper.ts
import { Department } from '../../domain/entities/department.entity';
import { DepartmentDto } from '../dto/department.dto';

export class DepartmentMapper {
  static toDto(department: Department): DepartmentDto {
    return new DepartmentDto({
      id: department.id,
      name: department.name,
      code: department.code,
      description: department.description,
      isActive: department.isActive,
      createdAt: department.createdAt,
      updatedAt: department.updatedAt,
    });
  }

  static toDtoList(departments: Department[]): DepartmentDto[] {
    return departments.map((department) => DepartmentMapper.toDto(department));
  }
}

File: backend-spring/application/src/main/java/apptemplate/application/mappers/DepartmentMapper.java
package apptemplate.application.mappers;

import apptemplate.application.dto.department.DepartmentDto;
import apptemplate.domain.entities.Department;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper(componentModel = "spring")
public interface DepartmentMapper {

    DepartmentMapper INSTANCE = Mappers.getMapper(DepartmentMapper.class);

    @Mapping(target = "isActive", source = "active")
    DepartmentDto toDto(Department department);
}
